config/git_fetcher.go

```diff
@@ -148,6 +149,23 @@ func (g *GitFetcher) All() map[string][]string {
 	return newmap
 }
 
+func (g *GitFetcher) caseFoldKey(key string) string {
+	parts := strings.Split(key, ".")
+	last := len(parts) - 1
+
+	// We check for 3 or more parts here because if the middle part is a
+	// URL, it may have dots in it. We'll downcase the part before the first
+	// dot and after the last dot, but preserve the piece in the middle,
+	// which may be a branch name, remote, or URL, all of which are
+	// case-sensitive. This is the algorithm Git uses to canonicalize its
+	// keys.
+	if len(parts) < 3 {
+		return strings.ToLower(key)
+	}
+
+	return fmt.Sprintf("%s.%s.%s", strings.ToLower(parts[0]), strings.Join(parts[1:last], "."), strings.ToLower(parts[last]))
```

Comment:
One alternative option here (which might--or might not be--be clearer than `%s.%s.%s`) is:

```go
return strings.Join([]string{
        strings.ToLower(parts[0]),
        strings.Join(parts[1:last], "."),
        strings.ToLower(parts[last]),
}, ".")
```

But, I'm not convinced that one is better than the other overall. Totally your call on which you'd like to adopt.

Reply:
Yeah, that would in fact be tidier, I think. I'll make that change.